admin/main.go

```diff
@@ -184,6 +185,176 @@ func verifyIDToken(app *firebase.App, idToken string) *auth.Token {
+// https://firebase.google.com/docs/auth/admin/manage-users
+// ==================================================================
+
+func getUser(ctx context.Context, client *auth.Client) *auth.UserRecord {
+	uid := "some_string_uid"
+
+	// [START get_user]
+	u, err := client.GetUser(ctx, uid)
+	if err != nil {
+		log.Fatalf("error getting user %s: %v\n", uid, err)
+	}
+	log.Printf("Successfully fetched user data: %v\n", u)
+	// [END get_user]
+	return u
+}
+
+func getUserByEmail(ctx context.Context, client *auth.Client) *auth.UserRecord {
+	email := "some@email.com"
+	// [START get_user_by_email]
+	u, err := client.GetUserByEmail(ctx, email)
+	if err != nil {
+		log.Fatalf("error getting user by email %s: %v\n", email, err)
+	}
+	log.Printf("Successfully fetched user data: %v\n", u)
+	// [END get_user_by_email]
+	return u
+}
+
+func getUserByPhone(ctx context.Context, client *auth.Client) *auth.UserRecord {
+	phone := "+13214567890"
+	// [START get_user_by_phone]
+	u, err := client.GetUserByPhoneNumber(ctx, phone)
+	if err != nil {
+		log.Fatalf("error getting user by phone %s: %v\n", phone, err)
+	}
+	log.Printf("Successfully fetched user data: %v\n", u)
+	// [END get_user_by_phone]
+	return u
+}
+
+func createUser(ctx context.Context, client *auth.Client) *auth.UserRecord {
+	// [START create_user]
+	u, err := client.CreateUser(context.Background(),
+		(&auth.UserToCreate{}).
+			Email("user@example.com").
+			EmailVerified(false).
+			PhoneNumber("+15555550100").
+			Password("secretPassword").
+			DisplayName("John Doe").
+			PhotoURL("http://www.example.com/12345678/photo.png").
+			Disabled(false))
+
+	if err != nil {
+		log.Fatalf("error creating user: %v\n", err)
+	}
+	log.Printf("Successfully created user: %v\n", u)
+	// [END create_user]
+	return u
+}
+
+func createUserWUID(ctx context.Context, client *auth.Client) *auth.UserRecord {
+	uid := "something"
+	// [START create_user_with_uid]
+	u, err := client.CreateUser(context.Background(),
+		(&auth.UserToCreate{}).UID(uid).Email("user@example.com").PhoneNumber("+15555550100"))
+	// alternatively
+	params := auth.UserToCreate{}
+	// or: 	var params auth.UserToCreate
+	u2, err := client.CreateUser(context.Background(),
+		params.UID(uid).Email("user@example.com").PhoneNumber("+15555550100"))
+	// alternatively
+
+	if err != nil {
+		log.Fatalf("error creating user: %v\n", err)
+	}
+	log.Printf("Successfully created user: %v\n", u)
+	// [END create_user_with_uid]
+	return u2
+}
+
+func updateUser(ctx context.Context, client *auth.Client) *auth.UserRecord {
+	uid := "d"
+	// [START update_user]
+	u, err := client.UpdateUser(context.Background(), uid,
+		(&auth.UserToUpdate{}).
+			Email("user@example.com").
+			EmailVerified(true).
+			PhoneNumber("+15555550100").
+			Password("newPassword").
+			DisplayName("John Doe").
+			PhotoURL("http://www.example.com/12345678/photo.png").
+			Disabled(true))
+	// see "CreateUser, above for syntax alteratives to (&auth.UserTo...{})"
```

Comment:
I'd remove this comment since we don't know that this code snippet will always follow the previous one in docs.